device/common/include/traccc/fitting/device/impl/fit.ipp

```diff
@@ -57,7 +57,8 @@ TRACCC_HOST_DEVICE inline void fit(
                                           *(seqs_view.ptr() + param_id));
 
     // Run fitting
-    kalman_fitter_status fit_status = fitter.fit(seed_param, fitter_state);
+    [[maybe_unused]] kalman_fitter_status fit_status =
+        fitter.fit(seed_param, fitter_state);
```

Comment:
I believe this is because for some backends (HIP if I'm remembering correctly? @StewMH ?) `assert` isn't defined, so we don't actually end up using this `fit_status` variable.

Reply:
In an opt build assert isn't called so arguably it is unused here. There is a different weird thing with HIP assert: https://github.com/acts-project/algebra-plugins/pull/145